newIDE/app/src/MainFrame/index.js

```diff
@@ -1353,21 +1374,21 @@ class MainFrame extends React.Component<Props, State> {
     }
   };
 
-  _onProjectItemModified = () => {
-    if (this.props.unsavedChanges)
-      this.props.unsavedChanges.triggerUnsavedChanges();
-    this.forceUpdate();
+  const _onProjectItemModified = () => {
+    if (props.unsavedChanges) props.unsavedChanges.triggerUnsavedChanges();
+    forceUpdate();
   };
 
-  _onCreateEventsFunction = (
+  const _onCreateEventsFunction = (
     extensionName: string,
-    eventsFunction: gdEventsFunction
+    eventsFunction: gdEventsFunction,
+    newState: State = state
   ) => {
-    const { currentProject } = this.state;
+    const { currentProject } = newState;
     if (!currentProject) return;
-    const { eventsFunctionsExtensionsState } = this.props;
+    const { eventsFunctionsExtensionsState } = props;
```

Comment:
I think you're using the "old props" here, sounds dangerous.
**Don't pass newProps as argument**. Instead, just pass `eventsFunctionsExtensionsState` (by passing `newProps.eventsFunctionsExtensionsState`).

The idea is to pass to the function as little as it needs.

Reply:
I didn't realise closure is going to be a problem for props too, until I discovered what happened with the unsavedChanges.

Yes, you're right about this, I need to figure out if stale props are going to be a problem in the entire code, not just this particular place.